env/env.go

```diff
@@ -88,20 +79,30 @@ func PrepareHelp(config interface{}) (string, error) {
 	var helpAux []byte
 	configValue := reflect.ValueOf(config).Elem()
 	for i := 0; i < configValue.NumField(); i++ {
-		fieldName := configValue.Type().Field(i).Name
-		var snakeCase []byte
-		for i, c := range fieldName {
-			if i > 0 && c >= 'A' && c <= 'Z' {
-				snakeCase = append(snakeCase, '_')
-			}
-			snakeCase = append(snakeCase, byte(c))
+		confKey := getConfKey(configValue.Type().Field(i))
+		if confKey == "-" {
+			continue
 		}
+
 		prefix := ""
 		if goconfig.PrefixEnv != "" {
 			prefix = goconfig.PrefixEnv + "_"
 		}
-		helpAux = append(helpAux, []byte(prefix+strings.ToUpper(string(snakeCase)))...)
-		helpAux = append(helpAux, []byte("=value\n")...)
+		helpAux = append(helpAux, []byte(prefix+strings.ToUpper(confKey)+"=value\n")...)
 	}
 	return string(helpAux), nil
 }
+
+func getConfKey(field reflect.StructField) string {
+	confKey := field.Tag.Get("env")
```

Comment:
Just a note. You will be fine with a `key` name for a variable (no clash). In a `go`, it's perfectly normal to have even smaller names like `k` 🤷.

Reply:
Fair point. The func name should give enough clues about kind of key are dealing here with.